### PR TITLE
column layout option added

### PR DESCRIPTION
**Pull request by kirsty-hames**
- implementation based on GMCQ and Hotgrid _columns
- example _comment inconsistencies corrected

### CODE REVIEW

**Comment by kirsty-hames** [comment]
I've set up an instance in the adapt_kineo_theme svn, Kineo Presentation topic.

**Comment by guywillis** [comment]
Looking good, @kirsty-hames.

There's a few instances of white space that need removing from the .less (L129) and .hbs (L20, L22, L30, L38) files.

Also wanted to check that the mobile view is as intended. The hotgrid / gmcq mobile view sets each item width to 50% whereas the list is setting the item width to 100%.

No column mobile view
Screenshot 2020-10-21 at 10 08 40

Column mobile view
Screenshot 2020-10-21 at 10 09 11

**Comment by kirsty-hames** [comment]
Thanks @guywillis. I was thinking List items are generally more text heavy then Hotgrid or GMCQ so a two column layout would mean many lines of text. 
However, we probably wouldn't want the image to display larger than it does on desktop for mobile e.g 225px wide for 4 columns on desktop vs 500px on mobile. Perhaps inheriting the desktop width and centring or should we leave that for the theming?

**Comment by guywillis** [comment]
Good point, they usually do contain more text. 

Images in Adapt will always stretch to fill their container unless we override the default. Providing the images are created at the standard single width size (540px) the stretching should not be a problem. 

Upon reflection I think the proposed is a good solution - just the white space to remove please.

**Review by guywillis — APPROVED** [review]
👍

File: templates/list.hbs [hunk 1]
```diff
@@ -4,39 +4,41 @@
 
   <div class="component__widget list__widget">
 
-    <div class="list__container{{#if _orderedList}} is-ordered-list{{/if}}" role="list">
+    <div class="list__container{{#if _orderedList}} is-ordered-list{{/if}}{{#if _columns}} has-columns{{/if}}" role="list">
```
Comment (tomgreenfield) [inline templates/list.hbs:7]: Suggest only applying the column styling if `_columns` is set to be greater than `1`?
```hbs
<div class="list__container{{#if _orderedList}} is-ordered-list{{/if}}{{#compare _columns '>' 1}} has-columns{{/compare}}" role="list">
```

Think that's the correct use of the helper; might need a test.

File: js/list-view.js [hunk 2]
```diff
@@ -1,24 +1,41 @@
...
     },
 
+    resizeControl: function () {
+      this.setUpColumns();
+    },
+
+    setUpColumns: function () {
+      var columns = this.model.get('_columns');
+
+      if (!columns) return;
```
Comment (tomgreenfield) [inline js/list-view.js:33]: No need to set width if not enough columns:
```js
if (columns < 2) return;
```

**Comment by tomgreenfield** [comment]
Would you mind applying the suggestions and sorting the conflict when you're next free, @kirsty-hames?

Or I'm happy to step in if you'd prefer?

**Comment by kirsty-hames** [comment]
> Would you mind applying the suggestions and sorting the conflict when you're next free, @kirsty-hames?
> 
> Or I'm happy to step in if you'd prefer?

Thanks @tomgreenfield, if you have time to pick up that would be appreciated otherwise I'll pick this up when I'm next free.

**Comment by kirsty-hames** [comment]
Suggestions applied and conflict amended @tomgreenfield. Can you review when you have time please?

**Review by tomgreenfield — APPROVED** [review]
:eye:

**Pull request merged**